File: exercises/string_l2_uncompress/test_cases.py
test_cases = {
    # --------------------
    # Given tests
    # --------------------
    "test_00": {
        "input": ["2c3a1t"],
        "expected": "ccaaat",
    },
    "test_01": {
        "input": ["4s2b"],
        "expected": "ssssbb",
    },
    "test_02": {
        "input": ["2p1o5p"],
        "expected": "ppoppppp",
    },
    "test_03": {
        "input": ["3n12e2z"],
        "expected": "nnneeeeeeeeeeeezz",
    },
    "test_04": {
        "input": ["127y"],
        "expected": "y" * 127,
    },

    # --------------------
    # Edge cases
    # --------------------

    # edge_00: single group
    "edge_00": {
        "input": ["5a"],
        "expected": "aaaaa",
    },

    # edge_01: multiple single counts
    "edge_01": {
        "input": ["1a1b1c1d"],
        "expected": "abcd",
    },

    # edge_02: multi-digit numbers
    "edge_02": {
        "input": ["10x2y"],
        "expected": "x" * 10 + "yy",
    },

    # edge_03: large count
    "edge_03": {
        "input": ["50z"],
        "expected": "z" * 50,
    },

    # edge_04: letters mixed with digits in pattern
    "edge_04": {
        "input": ["2a3b4c"],
        "expected": "aabbbcccc",
    },

    # edge_05: single character result
    "edge_05": {
        "input": ["1q"],
        "expected": "q",
    },

    # edge_06: repeated different chars
    "edge_06": {
        "input": ["2a2b2c2d"],
        "expected": "aabbccdd",
    },
}

File: exercises/string_l2_compress/test_cases.php
<?php

return $testCases = [

    // --------------------
    // Given tests
    // --------------------
    "test_00" => [
        "input" => ["ccaaatsss"],
        "expected" => "2c3at3s",
    ],
    "test_01" => [
        "input" => ["ssssbbz"],
        "expected" => "4s2bz",
    ],
    "test_02" => [
        "input" => ["ppoppppp"],
        "expected" => "2po5p",
    ],
    "test_03" => [
        "input" => ["nnneeeeeeeeeeeezz"],
        "expected" => "3n12e2z",
    ],
    "test_04" => [
        "input" => [str_repeat("y", 127)],
        "expected" => "127y",
    ],

    // --------------------
    // Edge cases
    // --------------------

    // edge_00: single character
    "edge_00" => [
        "input" => ["a"],
        "expected" => "a",
    ],

    // edge_01: all unique characters
    "edge_01" => [
        "input" => ["abcdef"],
        "expected" => "abcdef",
    ],

    // edge_02: two characters same
    "edge_02" => [
        "input" => ["aa"],
        "expected" => "2a",
    ],

    // edge_03: alternating pattern
    "edge_03" => [
        "input" => ["ababab"],
        "expected" => "ababab",
    ],

    // edge_04: large block then singles
    "edge_04" => [
        "input" => ["aaaaab"],
        "expected" => "5ab",
    ],

    // edge_05: mixed sizes
    "edge_05" => [
        "input" => ["aaabbccccd"],
        "expected" => "3a2b4cd",
    ],

    // edge_06: very large count
    "edge_06" => [
        "input" => [str_repeat("z", 500)],
        "expected" => "500z",
    ],

    // edge_07: two large groups
    "edge_07" => [
        "input" => [str_repeat("a", 20) . str_repeat("b", 15)],
        "expected" => "20a15b",
    ],

];